dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/infer/DescriptorLanguageInferrer.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2024 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.helpers.infer;
+
+import io.dockstore.common.DescriptorLanguage;
+import io.dockstore.common.DescriptorLanguageSubclass;
+import io.dockstore.common.EntryType;
+import io.dockstore.webservice.CustomWebApplicationException;
+import io.dockstore.webservice.helpers.FileTree;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides the infrastructure for Inferrers that handle a particular descriptor language.
+ * See the comments in the `infer` method for an overview of the technique used.
+ * Concrete implementations can/should override various methods to implement language-specific behavior,
+ * most typically the `determineType`, `determineName`, and `determineSubclass` methods.
+ */
+public abstract class DescriptorLanguageInferrer implements Inferrer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DescriptorLanguageInferrer.class);
+    /**
+     * Regular expression that matches pathlike strings, composed of the currently-legal SourceFile path characters.
+     */
+    private static final Pattern POSSIBLE_PATH = Pattern.compile("[./]*+[a-zA-Z0-9/_-]++\\.[a-zA-Z0-9]++");
```

Comment:
This regex seems different from the configured default, should one reference the other anyway since it can change?

https://github.com/dockstore/dockstore/blob/develop/dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceConfiguration.java#L164
https://github.com/dockstore/dockstore/blob/develop/dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceApplication.java#L699

Reply:
Goal of the regexp is to identify strings that strongly resemble paths, to avoid accidentally detecting file references that aren't actually file references.  Depending on the language, paths can be quoted or or not, or delimited by spaces and other punctuation, so we don't include the space in the regexp.  As long as this regexp matches a subset of the paths allowable as SourceFile paths, and matches almost all of file references in real-world descriptor files, we're ok.  Probably meets those criteria as is, methinks.

Reply:
Let me re-phrase in a different way. So I thought the reason we made the set of allowable characters in filenames in the web.yml configurable was to allow us to add support for say filenames with with European accented characters or Chinese names, etc. instead of the default ASCII if we decided to. 

Why would we want to, say add support for one of those globally but not to inferring .dockstore.yml or vice versa? 
Is there a reason to not make these two patterns the same or at least linked via documentation in some-way? 

Otherwise, it seems that if we wanted to add support for European accented characters we'd need to change the web.yml (possible at runtime) but we'd still need a new build for the webservice, assuming future developer knew to change it in two places.

Reply:
I rejiggered the code that finds referenced paths, so now, it splits the file content into chunks using a regexp composed of the list of separator characters (colon, quotes, brackets, parentheses, etc).  No matter how we decide to configure the webservice, these chars will likely never be added to the SourceFile path regexp, so the code should work with whatever setting we choose.